### DISCUSSION TITLE
Fix the "Sharing" feature

### DISCUSSION THREAD

**Discussion by Anon666333** [immich-app/immich#14282] [Feature Request]
Starting off - HUGE immich fan. Cannot thank the dev team enough!

I've noticed that the "sharing" options are kind of inconsistent.
Specifically on the mobile app (where immich is used most). When looking at a photo, in the bottom left there is a "share" option that has the share symbol (reference A). However, this opens the options to share to other apps/text/etc.
If you then select the image from the photo grid (pressing and holding), that same icon, still called "share" now allows you to create a shareable link to the photo via immich. There is then a new button called "share to" with a different symbol to share to specific apps. (Reference B).

My suggestion, is to consolidate this all into a single "share" button, so you can share to another app or share a custom link within a single button from either within the picture itself or from selecting multiple images from the photo grid.

Thank you!!

**Reference A**
![Screenshot_20241121_082003.jpg](https://github.com/user-attachments/assets/dd87c869-e7a9-4dd2-8e4e-9bd44f2c2867)

![Screenshot_20241121_082118_IntentResolver.jpg](https://github.com/user-attachments/assets/9630795f-e1af-4462-857a-5e717a31120c)

**Reference B**
![Screenshot_20241121_082209.jpg](https://github.com/user-attachments/assets/e36a7939-0fdc-40db-a05f-f36b2a7859a4)

**Comment by NicholasFlamy** [comment]
I agree with you, this was previously discussed in #1863 but that issue was primarily focused on the Multi-Select View and that has been implemented. I broke out the current state of things and the request for link sharing from the Photo Viewer in #15434.
